docs/maintainers/iavl-v1-migration.md

```diff
@@ -0,0 +1,308 @@
+# IAVL v1 Migration Guide for Consensus Nodes
+
+This guide helps consensus node operators migrate from IAVL v0 to IAVL v1 to achieve optimal performance benefits.
+
+## Background
+
+Starting with celestia-app v4, the state machine uses IAVL v1.x instead of v0.x. IAVL v1 introduces data locality optimizations that provide roughly **10x performance improvement** over the previous version.
+
+### Key Benefits of IAVL v1
+
+- **Significantly faster state access**: ~10x performance improvement
+- **Better data locality**: More efficient disk I/O patterns
+- **Reduced state access bottlenecks**: Prevents state access from limiting throughput
```

Comment:
redo this and don't harp on the benefits that much, we only care about informing node operators what they need to do to switch. focus on that.